flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/org/apache/flink/cdc/connectors/postgres/table/PostgreSQLDeserializationConverterFactory.java

```diff
@@ -96,4 +101,36 @@ public Object convert(Object dbzObj, Schema schema) throws Exception {
                     }
                 });
     }
+
+    private static Optional<DeserializationRuntimeConverter> createArrayConverter() {
+        return Optional.of(
+            new DeserializationRuntimeConverter() {
+
+                private static final long serialVersionUID = 1L;
+
+                @Override
+                public Object convert(Object dbzObj, Schema schema) throws Exception {
+                    if (dbzObj == null) {
+                        return null;
+                    }
+                    if (Schema.Type.ARRAY.equals(schema.type()) && dbzObj instanceof List && schema.valueSchema() != null) {
+                        switch (schema.valueSchema().type()) {
```

Comment:
I didn't realize that pgsql array can only support these schema.types. There are more such as Map boleans and so on.